pyrefly/lib/test/subscript_narrow.rs

```diff
@@ -241,6 +241,35 @@ def use2(mapping: TD) -> None:
 "#,
 );
 
+testcase!(
+    test_typeddict_in_literal_key_narrow,
+    r#"
+from typing import TypedDict, NotRequired, assert_type
+
+class TD(TypedDict):
+    foo: NotRequired[int]
+
+def use(mapping: TD) -> None:
+    if "foo" in mapping:
+        assert_type(mapping["foo"], int)
+    else:
+        mapping["foo"]  # E: TypedDict key `foo` may be missing; guard this access with `'foo' in obj` or `obj.get('foo')`
```

Comment:
can you add a test case for `.get` as well, please? I don't remember if its behavior is special-cased.

Reply:
also, a test case for `if "foo" not in mapping` + accessing the key in else branch